### PR TITLE
Ignore empty path filters, disable cred provider on hosted, disable config creds on on-prem (M103)

### PR DESCRIPTION
**Pull request by jmyersmsft**

### CODE REVIEW

File: Tasks/Common/nuget-task-common/NuGetToolRunner.ts [hunk 1]
```diff
@@ -155,6 +156,78 @@ export function locateNuGetExe(userNuGetExePath: string): string {
...
+    if (credentialProviderOverrideFlag === "true") {
+        tl.debug("Credential provider is force-enabled for testing purposes.");
+        return true;
+    }
+
+    if (credentialProviderOverrideFlag === "false") {
+        tl.debug("Credential provider is force-disabled for testing purposes.");
+        return false;
+    }
+
```
Comment (bryanmacfarlane) [inline Tasks/Common/nuget-task-common/NuGetToolRunner.ts]: ??
let enabled = isHosted();
tl.debug("Credential provider enabled: " + enabled);
return enabled;

File: Tasks/Common/nuget-task-common/NuGetToolRunner.ts [hunk 2]
```diff
@@ -155,6 +156,78 @@ export function locateNuGetExe(userNuGetExePath: string): string {
     return toolPath;
 }
 
+function isHosted() {
+    // not an ideal way to detect hosted, but there isn't a variable for it, and we can't make network calls from here
+    // due to proxy issues.
+    const collectionUri = tl.getVariable("System.TeamFoundationCollectionUri");
+    const isHosted = /\.visualstudio\.com$/i.test(collectionUri);
```
Comment (bryanmacfarlane) [inline Tasks/Common/nuget-task-common/NuGetToolRunner.ts]: doesn't this need to return isHosted?

File: Tasks/Common/nuget-task-common/NuGetToolRunner.ts [hunk 3]
```diff
@@ -155,6 +156,78 @@ export function locateNuGetExe(userNuGetExePath: string): string {
     return toolPath;
 }
 
+function isHosted() {
```
Comment (bryanmacfarlane) [inline Tasks/Common/nuget-task-common/NuGetToolRunner.ts]: function isHosted(): Boolean

File: Tasks/Common/nuget-task-common/NuGetToolRunner.ts (hunk 1 above)
Comment (jmyersmsft) [inline Tasks/Common/nuget-task-common/NuGetToolRunner.ts]: no, we want the override flag to be able to force the decision either way

File: Tasks/Common/nuget-task-common/NuGetToolRunner.ts (hunk 2 above)
Comment (jmyersmsft) [inline Tasks/Common/nuget-task-common/NuGetToolRunner.ts]: whoops, yes

**Comment by daazose** [comment]
Signed off

**Pull request merged**